packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -7,37 +7,64 @@ import { get } from 'lodash';
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
-import { Button } from '@wordpress/components';
+import { Button, Icon } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 import { wordpress } from '@wordpress/icons';
 
 function FullscreenModeClose() {
-	const { isActive, postType } = useSelect( ( select ) => {
-		const { getCurrentPostType } = select( 'core/editor' );
-		const { isFeatureActive } = select( 'core/edit-post' );
-		const { getPostType } = select( 'core' );
+	const { isActive, isRequestingSiteIcon, postType, siteIconUrl } = useSelect(
+		( select ) => {
+			const { getCurrentPostType } = select( 'core/editor' );
+			const { isFeatureActive } = select( 'core/edit-post' );
+			const { isResolving } = select( 'core/data' );
+			const { getEntityRecord, getPostType } = select( 'core' );
+			const siteData =
+				getEntityRecord( 'root', '__unstableBase', undefined ) || {};
 
-		return {
-			isActive: isFeatureActive( 'fullscreenMode' ),
-			postType: getPostType( getCurrentPostType() ),
-		};
-	}, [] );
+			return {
+				isActive: isFeatureActive( 'fullscreenMode' ),
+				isRequestingSiteIcon: isResolving( 'core', 'getEntityRecord', [
+					'root',
+					'__unstableBase',
+					undefined,
+				] ),
+				postType: getPostType( getCurrentPostType() ),
+				siteIconUrl: siteData.site_icon_url,
+			};
+		},
+		[]
+	);
 
 	if ( ! isActive || ! postType ) {
 		return null;
 	}
 
+	let buttonIcon = <Icon size="36px" icon={ wordpress } />;
+
+	if ( siteIconUrl ) {
+		buttonIcon = (
+			<img
+				alt="site-icon"
```

Comment:
Bit of a nitpick, but shouldn't the [`alt` text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) be more human- (rather than machine-) readable? (This might imply we need to translate it.)

E.g.

```suggestion
				alt={ __( 'Site Icon' ) }
```